lib/visitors_manipulator.rb

```diff
@@ -24,17 +24,22 @@ def update
     go_forward = visit.visitors.all?(&:valid?) && visit.valid?(:visitors_set)
 
     if params[:next] == 'Add another visitor'
-      if visit.visitors.size < Visit::MAX_VISITORS
-        visit.visitors << model_class.new
-        redirect_to this_path
-      else
-        redirect_to this_path, notice: "You may only have a maximum of #{Visit::MAX_VISITORS} visitors"
-      end
+      max_visitors
     else
       redirect_to go_forward ? next_path : this_path
     end
   end
 
+  def max_visitors
+    if visit.visitors.size < Visit::MAX_VISITORS
+      visit.visitors << model_class.new
+      redirect_to this_path
+    else
+      flash[:notice] = I18n.t(:max_visitors, scope: 'controllers.shared')
+      redirect_to this_path
+    end
+  end
```

Comment:
I love that you've extracted this, but I think the name is misleading. It seems to do two things:

1. ensure that there is a blank visitor to add a new one, if you haven't reached the maximum, or
2. flash a notice if you have reached the maximum, then
3. redirect.

But as it always redirects identically, perhaps that doesn't belong in the method. Then this method would be something like:

```ruby
def prepare_next_visitor
  if visit.visitors.size < Visit::MAX_VISITORS
    visit.visitors << model_class.new
  else
    flash[:notice] = I18n.t(:max_visitors, scope: 'controllers.shared')
  end
end
```

and we'd call redirect after returning.